### PR TITLE
Revert removing `requests` and make it a hard dependency instead

### PR DESCRIPTION
**Pull request by pmeier**
Addresses https://github.com/pytorch/vision/pull/4973#issuecomment-987706250.

cc @pmeier

### CODE REVIEW

File: .circleci/config.yml.in [hunk 1]
```diff
@@ -309,6 +309,9 @@ jobs:
       - pip_install:
           args: mypy
           descr: Install Python type check utilities
+      - run:
+          name: Install library stub packages
+          command: mypy --install-types
```
Comment (pmeier) [inline .circleci/config.yml.in:314]: Although we currently only need [types-requests](https://pypi.org/project/types-requests/), this is a convenience command to autodetect all missing library stubs and install them.

**Comment by datumbox** [comment]
There are some related [failures](https://app.circleci.com/pipelines/github/pytorch/vision/13160/workflows/96f41d10-d6eb-4048-8d44-b11656ce88f3/jobs/1050593). Feel free to ping me when you get a green CI to review.

**Review by datumbox — APPROVED** [review]
LGTM, feel free to merge on green CI.

We should also add the dependency on the draft PR you started to organize the dependencies per module.

**Comment by fmassa** [comment]
@pmeier note that it is most probably possible to implement the needed functionality relying only on `urllib`.

Let's keep this in mind for the future.

**Pull request merged**